c8run/internal/archive/archive.go

```diff
@@ -0,0 +1,289 @@
+package archive
+
+import (
+	"archive/tar"
+	"archive/zip"
+	"compress/gzip"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+func DownloadFile(filepath string, url string) error {
+	out, err := os.Create(filepath)
+	if err != nil {
+		return err
+	}
+	defer out.Close()
+
+	resp, err := http.Get(url)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("bad status: %s", resp.Status)
+	}
+
+	_, err = io.Copy(out, resp.Body)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func CreateTarGzArchive(files []string, buf io.Writer) error {
+	gw := gzip.NewWriter(buf)
+	defer gw.Close()
+	tw := tar.NewWriter(gw)
+	defer tw.Close()
+
+	for _, file := range files {
+		err := filepath.Walk(file, func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if !info.IsDir() {
+				return addToArchive(tw, path)
+			} else {
+				// Add directory to the archive
+				header, err := tar.FileInfoHeader(info, path)
+				if err != nil {
+					return err
+				}
+				header.Name = path + "/"
+				if err := tw.WriteHeader(header); err != nil {
+					return err
+				}
+				return nil
+
+			}
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func ExtractTarGzArchive(filename string, xpath string) error {
+	tarFile, err := os.Open(filename)
+	if err != nil {
+		panic(err)
+	}
+	defer tarFile.Close()
+
+	gz, err := gzip.NewReader(tarFile)
+	if err != nil {
+		panic(err)
+	}
+	defer gz.Close()
+
+	absPath, err := filepath.Abs(xpath)
+
+	_, err = os.Stat(absPath)
+	if errors.Is(err, os.ErrNotExist) {
+		os.Mkdir(absPath, 0755)
+	}
+
+	tr := tar.NewReader(gz)
+	for {
+		hdr, err := tr.Next()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			panic(err)
```

Comment:
why panic ?

Reply:
I've removed my panics to just main and stubs.
https://github.com/camunda/camunda/pull/24040/commits/4aa3968f155af3b001d67e383aedef54f6eb22af